src/WPF/ArcGISRuntime.WPF.Samples/Samples/Data/EditAndSyncFeatures/EditAndSyncFeatures.xaml.cs

```diff
@@ -0,0 +1,511 @@
+
+            // Return if the extent overlay is null
+            if (extentOverlay == null) { return; }
+
+            // Get the extent graphic
+            Graphic extentGraphic = extentOverlay.Graphics.FirstOrDefault();
+
+            // Create the extent graphic and add it to the overlay if it doesn't exist
+            if (extentGraphic == null)
+            {
+                extentGraphic = new Graphic(envelopeBldr.ToGeometry());
+                extentOverlay.Graphics.Add(extentGraphic);
+            }
+            else
+            {
+                // Otherwise, simply update the graphic's geometry
+                extentGraphic.Geometry = envelopeBldr.ToGeometry();
+            }
+        }
+
+        private async void StartGeodatabaseGeneration()
+        {
+            // Create a task for generating a geodatabase (GeodatabaseSyncTask)
+            _gdbSyncTask = await GeodatabaseSyncTask.CreateAsync(_featureServiceUri);
+
+            // Get the current extent of the red preview box
+            Envelope extent = MyMapView.GraphicsOverlays.FirstOrDefault().Extent as Envelope;
+
+            // Get the default parameters for the generate geodatabase task
+            GenerateGeodatabaseParameters generateParams = await _gdbSyncTask.CreateDefaultGenerateGeodatabaseParametersAsync(extent);
+
+            // Create a generate geodatabase job
+            GenerateGeodatabaseJob _generateGdbJob = _gdbSyncTask.GenerateGeodatabase(generateParams, _gdbPath);
+
+            // Handle the job changed event
+            _generateGdbJob.JobChanged += GenerateGdbJobChanged;
+
+            // Handle the progress changed event (to show progress bar)
+            _generateGdbJob.ProgressChanged += ((object sender, EventArgs e) =>
+            {
+                // Get the job
+                GenerateGeodatabaseJob job = sender as GenerateGeodatabaseJob;
+
+                // Update the progress bar
+                UpdateProgressBar(job.Progress);
+            });
+
+            // Start the job
+            _generateGdbJob.Start();
+        }
+
+        private async void HandleGenerationStatusChange(GenerateGeodatabaseJob job)
+        {
+            JobStatus status = job.Status;
+
+            // If the job completed successfully, add the geodatabase data to the map
+            if (status == JobStatus.Succeeded)
+            {
+                // Clear out the existing layers
+                MyMapView.Map.OperationalLayers.Clear();
+
+                // Get the new geodatabase
+                _resultGdb = await job.GetResultAsync();
+
+                // Loop through all feature tables in the geodatabase and add a new layer to the map
+                foreach (GeodatabaseFeatureTable table in _resultGdb.GeodatabaseFeatureTables)
+                {
+                    // Create a new feature layer for the table
+                    FeatureLayer _layer = new FeatureLayer(table);
```

Comment:
Variables that begin with an underscore ( _XXXXX ) should be declared as global in scope. In this case you have a local variable and it should be names something like: myFeatureLayer not _layer.